### DISCUSSION TITLE
On working with partial types

### DISCUSSION THREAD

**Discussion by Allaoua9** [prisma/prisma#10273] [Q&A] [answered]
In the documentation about advanced type safety, a solution is proposed for using variations of the generated model type. It is recommended to use the `Prisma.validator`

```ts
import { Prisma } from '@prisma/client'

// 1: Define a type that includes the relation to `Post`
const userWithPosts = Prisma.validator<Prisma.UserArgs>()({
  include: { posts: true },
})

// 2: Define a type that only contains a subset of the scalar fields
const userPersonalData = Prisma.validator<Prisma.UserArgs>()({
  select: { email: true, name: true },
})

// 3: This type will include a user and all their posts
type UserWithPosts = Prisma.UserGetPayload<typeof userWithPosts>
```

I find this solution hard to read.
My question is, why not just use raw typescript for this ?

```ts
type UserWithPosts = User & { posts: Post[] }

type UserPersonalData = Pick<User, 'email' | 'name'>
```

**Comment by austincrim** [comment, marked as answer]
Hi @Allaoua9 👋, thanks for the question.

The Prisma type helpers are a great way to generate custom types using the same syntax as Prisma Client queries. The are especially valuable when including many nested relations or selecting subsets of large models.

You can use the type helpers directly without the use of `Prisma.validator`.

```ts
type UserWithPosts = Prisma.UserGetPayload<{
  include: { posts: true } 
}>
```

That being said, you can create your custom types in whatever way you prefer, the Prisma helpers, regular TypeScript, or some combination of both.

Let me know if you have more specific questions!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
